README.md

```diff
@@ -65,9 +71,58 @@ A successful response object always contains data which can be iterated upon. Th
 response.data.invoice.line_items.first.price
 ```
 
-### Use Graphlient::Query directly
+You can also execute mutations the same way.
+
+```ruby
+response = client.query do
+  mutation do
+    createInvoice(input: { fee_in_cents: 12_345 }) do
+      id
+      fee_in_cents
+    end
+  end
+end
+```
+
+The successful response contains data in `response.data`. The following example returns the newly created invoice's ID.
+
+```ruby
+response.data.create_invoice.first.id
+```
+
+### Executing Parameterized Queries and Mutations
+
+Graphlient can execute parameterized queries and mutations by providing variables as query parameters.
+
+The following query accepts an array of IDs.
+
+```ruby
+client.query(ids: [42]) do
+  query(:$ids => :'[Int]') do
```

Comment:
oo interesting approach, I wonder if we can make this slightly more DSLy by doing things like:

``` 
client.query(ids: [42]) do
  query(:ids => [:int]) do
   ....
   end
end
```
This means on our end we need to detect query variables and add `$` to them when generating query. What do you think?

Reply:
Good idea, opened https://github.com/ashkan18/graphlient/issues/23 for a future improvement.